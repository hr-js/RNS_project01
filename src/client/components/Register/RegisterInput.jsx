// @flow
import * as React from "react";
import Input from "@components/Register/Input";
import SingleButton from "@components/SingleButton";
import styles from "./style.css";

type RegisterInputType = {
  firstNameValue: string,
  firstNameError: string,
  lastNameValue: string,
  lastNameError: string,
  emailValue: string,
  emailError: string,
  onFirstNameChange: (e: SyntheticInputEvent<HTMLInputElement>) => {},
  onLastNameChange: (e: SyntheticInputEvent<HTMLInputElement>) => {},
  onEmailChange: (e: SyntheticInputEvent<HTMLInputElement>) => {},
  onClickNextButton: (e: SyntheticEvent<HTMLButtonElement>) => {},
  onSubmitForm: (e: SyntheticEvent<HTMLFormElement>) => {}
};

function RegisterInput(props: RegisterInputType): React.Node {
  const {
    firstNameValue,
    firstNameError,
    lastNameValue,
    lastNameError,
    emailValue,
    emailError,
    onFirstNameChange,
    onLastNameChange,
    onEmailChange,
    onClickNextButton,
    onSubmitForm
  } = props;
  const { main, inputName, inputEmail, domain, contentWithButton } = styles;
  const domainName = process.env.E_MAIL_DOMAIN || "hoge.com";

  return (
    <div className={main}>
      <form onSubmit={onSubmitForm}>
        <div className={contentWithButton}>
          <div className={inputName}>
            <Input
              id="firstName"
              label="姓"
              error={firstNameError}
              value={firstNameValue}
              onChange={onFirstNameChange}
            />
            <Input
              id="lastName"
              label="名"
              error={lastNameError}
              value={lastNameValue}
              onChange={onLastNameChange}
            />
          </div>
          <div className={inputEmail}>
            <Input
              id="email"
              label="Email"
              error={emailError}
              value={emailValue}
              onChange={onEmailChange}
            />
            <span className={domain}>@{domainName}</span>
          </div>
        </div>
        <SingleButton
          text="次へ"
          className="next"
          onButtonClick={onClickNextButton}
        />
      </form>
    </div>
  );
}
export default RegisterInput;
