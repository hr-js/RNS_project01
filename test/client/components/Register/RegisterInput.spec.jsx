import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import RegisterInput from "@components/Register/RegisterInput";

describe("RegisterInput.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const fn = () => {};
      const tree = renderer.create(
        <RegisterInput
          firstNameValue="firstName"
          firstNameError=""
          lastNameValue="lastName"
          lastNameError=""
          emailValue="email"
          emailError=""
          onFirstNameChange={fn}
          onLastNameChange={fn}
          onEmailChange={fn}
          onClickNextButton={fn}
          onSubmitForm={fn}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    delete process.env.E_MAIL_DOMAIN;
    /**
     * testする内容
     * - propsに指定された値, 関数が想定通りに表示されている、渡されている
     */
    const onFirstNameChange = () => {};
    const onLastNameChange = () => {};
    const onEmailChange = () => {};
    const onClickNextButton = () => {};
    const submitFunc = jest.fn();

    const registerInput = shallow(
      <RegisterInput
        firstNameError="firstNameError"
        firstNameValue="firstName"
        onFirstNameChange={onFirstNameChange}
        lastNameError="lastNameError"
        lastNameValue="lastName"
        onLastNameChange={onLastNameChange}
        emailError="emailError"
        emailValue="email"
        onEmailChange={onEmailChange}
        onClickNextButton={onClickNextButton}
        onSubmitForm={submitFunc}
      />
    );

    it("form要素のonSubmitに、onSubmitFormに指定した関数を渡している", () => {
      expect(registerInput.find("form").prop("onSubmit")).toBe(submitFunc);
    });

    describe("Inputコンポーネント(firstName)", () => {
      const firstName = registerInput.find("#firstName");

      it("errorに、firstNameErrorで指定された値が渡されている", () => {
        expect(firstName.prop("error")).toBe("firstNameError");
      });

      it("valueに、firstNameValueで指定された値が渡されている", () => {
        expect(firstName.prop("value")).toBe("firstName");
      });

      it("onChangeに、onFirstNameChangeで指定された関数が渡されている", () => {
        expect(firstName.prop("onChange")).toBe(onFirstNameChange);
      });
    });

    describe("Inputコンポーネント(lastName)", () => {
      const lastName = registerInput.find("#lastName");

      it("errorに、lastNameErrorで指定された値が渡されている", () => {
        expect(lastName.prop("error")).toBe("lastNameError");
      });

      it("valueに、lastNameValueで指定された値が渡されている", () => {
        expect(lastName.prop("value")).toBe("lastName");
      });

      it("onChangeに、onLastNameChangeで指定された関数が渡されている", () => {
        expect(lastName.prop("onChange")).toBe(onLastNameChange);
      });
    });

    describe("Inputコンポーネント(email)", () => {
      const email = registerInput.find("#email");

      it("errorに、emailErrorで指定された値が渡されている", () => {
        expect(email.prop("error")).toBe("emailError");
      });

      it("valueに、emailValueで指定された値が渡されている", () => {
        expect(email.prop("value")).toBe("email");
      });

      it("onChangeに、onEmailChangeで指定された関数が渡されている", () => {
        expect(email.prop("onChange")).toBe(onEmailChange);
      });

      it("process.env.E_MAIL_DOMAINが設定されていない時、デフォルト値が表示される", () => {
        expect(registerInput.find(".domain").text()).toBe("@hoge.com");
      });

      it("process.env.E_MAIL_DOMAINが設定されている時、設定された値が表示される", () => {
        process.env.E_MAIL_DOMAIN = "test.com";

        const fn = () => {};
        const domain = shallow(
          <RegisterInput
            firstNameError=""
            firstNameValue=""
            onFirstNameChange={fn}
            lastNameError=""
            lastNameValue=""
            onLastNameChange={fn}
            emailError=""
            emailValue=""
            onEmailChange={fn}
            onClickNextButton={fn}
            onSubmitForm={fn}
          />
        )
          .find(".domain")
          .text();

        expect(domain).toBe("@test.com");
      });
    });

    it("SingleButtonをクリックした時、form要素のonSubmitに渡した関数が呼ばれる", () => {
      registerInput.find("form").simulate("submit");
      expect(submitFunc).toBeCalled();
    });
  });
});
