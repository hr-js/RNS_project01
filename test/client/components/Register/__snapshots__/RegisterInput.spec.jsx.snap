// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterInput.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="main"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="contentWithButton"
    >
      <div
        className="inputName"
      >
        <div
          className="inputArea"
        >
          <label
            htmlFor="firstName"
          >
            姓
            <span
              className="message"
            >
              
            </span>
          </label>
          <input
            className="input"
            id="firstName"
            onChange={[Function]}
            type="text"
            value="firstName"
          />
        </div>
        <div
          className="inputArea"
        >
          <label
            htmlFor="lastName"
          >
            名
            <span
              className="message"
            >
              
            </span>
          </label>
          <input
            className="input"
            id="lastName"
            onChange={[Function]}
            type="text"
            value="lastName"
          />
        </div>
      </div>
      <div
        className="inputEmail"
      >
        <div
          className="inputArea"
        >
          <label
            htmlFor="email"
          >
            Email
            <span
              className="message"
            >
              
            </span>
          </label>
          <input
            className="input"
            id="email"
            onChange={[Function]}
            type="text"
            value="email"
          />
        </div>
        <span
          className="domain"
        >
          @
          hoge.com
        </span>
      </div>
    </div>
    <div
      className="singleButton"
    >
      <button
        className="next"
        onClick={[Function]}
      >
        次へ
      </button>
    </div>
  </form>
</div>
`;
